### PR TITLE
Add performance operator upgrade test 4.4.0->4.5.0

### PR DESCRIPTION
**Pull request by dshchedr**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -22,6 +22,7 @@ require (
 	golang.org/x/sys v0.0.0-20200413165638-669c56c373c4 // indirect
 	golang.org/x/xerrors v0.0.0-20191204190536-9bdfabe68543 // indirect
 	k8s.io/api v0.18.0
+	k8s.io/apiextensions-apiserver v0.18.0
```
Comment (slintes) [inline go.mod:25]: please add this to the 1st `replace` section, so we use the same k8s version everywhere

**Comment by slintes** [comment]
the upgrade test failed, but it's green...?

**Comment by slintes** [comment]
1. ```  Expected
      <v1alpha1.ClusterServiceVersionPhase>: Installing
  to equal
      <v1alpha1.ClusterServiceVersionPhase>: Succeeded

  /go/src/github.com/openshift-kni/performance-addon-operators/functests-extended/1_performance_operator_upgrade/upgrade_operator.go:55```

2. and the cleanup failed, waiting for the MCP (we do not really need the cleanup, on the other hand it's nice to have it tested as well.

**Comment by dshchedr** [comment]
> the upgrade test failed, but it's green...?

not sure why it happens

**Comment by dshchedr** [comment]
> ```
>   Expected
>       <v1alpha1.ClusterServiceVersionPhase>: Installing
>   to equal
>       <v1alpha1.ClusterServiceVersionPhase>: Succeeded
> 
>   /go/src/github.com/openshift-kni/performance-addon-operators/functests-extended/1_performance_operator_upgrade/upgrade_operator.go:55
> ```
  Modified code - now it waits for Phase Succeeded first.
 
> and the cleanup failed, waiting for the MCP (we do not really need the cleanup, on the other hand it's nice to have it tested as well)
 Increased timeout to 30 minutes (same as we have in deploying script), let's look on result

**Comment by dshchedr** [comment]
hm, failed again.. looks like some race condition..
modified code again: combined waiting of Phase and ConditionReason to one method.

Unfortunatelly, can't check it locally now, so let's wait for another CI try

**Pull request merged**